### PR TITLE
ESQL: Preserve the projection of an aggregation

### LINKED ISSUE #99782: ESQL: KEEP after STATS trips assertion error
### Elasticsearch Version

main

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

all

### Problem Description

Do this

### Steps to Reproduce

```
rm /tmp/evil
for a in {0..9}; do
   for b in {0..9}; do
      echo '{"index": {}}' >> /tmp/evil
      echo '{"a": '$a', "b": '$b'}' >> /tmp/evil
   done
done
echo >> /tmp/evil

curl -uelastic:password -XDELETE localhost:9200/test
curl -HContent-Type:application/json -uelastic:password -XPUT localhost:9200/test
curl -HContent-Type:application/json -uelastic:password -XPOST localhost:9200/test/_bulk --data-binary @/tmp/evil
curl -HContent-Type:application/json -uelastic:password -XPOST localhost:9200/test/_forcemerge?max_num_segments=1
curl -HContent-Type:application/json -uelastic:password -XPOST localhost:9200/test/_refresh

curl -HContent-Type:application/json -uelastic:password -XPOST localhost:9200/_esql?pretty -d'{
   "query": "FROM test | STATS MAX(a) BY a, b | KEEP a"
}'
```

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by costin**
The internal AggregateOperators are focused creating groups and
 performing the aggregations using their intermediate states. However
 the logical projection is not enforced leading to errors downstream.
 Currently this is handled through OutputOperator which forces an
 alignment between the layout and the attributes however this is not
 only fragile but also insufficient.

This commit fixes a couple of things:
- introduces a ProjectOperator in front of the AggregateOperator. This is much easier, at least at the moment as it keeps the agg logic untouched.
- improves the ProjectOperator by moving away from the BitSet to a List that handles not just the selection but also ordering

Fix #99782

### CODE REVIEW

**Comment by costin** [comment]
This should also fix #99888

**Review by alex-spies — COMMENTED** [review]
This generally LGTM and could be merged without much change if it unblocks us.

However, since the local execution planner now plans projections differently, it would be nice to have unit tests that make sure we cover all weird edge cases that could happen during planning; That includes the newly supported double projections and reorderings.

Other than that I have mostly minor remarks.

File: x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec [hunk 1]
```diff
@@ -501,3 +523,5 @@ from employees | limit 10 | eval x = 1 | stats c = count(x);
 c:l
 10
 ;
```
[the remark this answers is not in the document]
Comment (astefan) [reply x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec:525]: ++

**Review by astefan — APPROVED** [review]
Other than what @alex-spies mentioned about the additional test covering https://github.com/elastic/elasticsearch/issues/99888 fix, LGTM.

**Review by luigidellaquila — APPROVED** [review]
LGTM too, +1 to @alex-spies suggestions

**Pull request merged**